client/api_client.go

```diff
@@ -61,8 +61,8 @@ type ApiClientInterface interface {
 	TeamCreate(payload TeamCreatePayload) (Team, error)
 	TeamUpdate(id string, payload TeamUpdatePayload) (Team, error)
 	TeamDelete(id string) error
-	TeamProjectAssignmentCreateOrUpdate(payload TeamProjectAssignmentPayload) (TeamProjectAssignment, error)
-	TeamProjectAssignmentDelete(assignmentId string) error
+	TeamProjectAssignmentCreateOrUpdate(payload *TeamProjectAssignmentPayload) (*TeamProjectAssignment, error)
+	TeamProjectAssignmentDelete(projectId string, teamId string) error
```

Comment:
I think that in "api_client" we should have only 3 methods, that serve all scopes for `team` (and additional 3 methods for `users`).
So each method will get the scopeId as an argument (aka `environmentId`, `projectId`, or `organizationId`. They are optional and one of them should exist).
That's because all of those are the same API and routes.

meaning `TeamProjectAssignmentCreateOrUpdate` and `TeamProjectAssignmentDelete` should be removed. And we should not "drag" / keep the existence of the old deprecated APIs names.

WDYS?

Reply:
the only scopes we haven't refactored yet - should preserve their methods here. Until we will refactor them too.
that's include:
1. org level user assignments
2. project level user assignments

Reply:
will be done in a follow-up PR.